lib/active_record/connection_adapters/postgis_adapter/rails3/spatial_column_info.rb

```diff
@@ -0,0 +1,42 @@
+module ActiveRecord  # :nodoc:
+  module ConnectionAdapters  # :nodoc:
+    module PostGISAdapter
+      # Do spatial sql queries for column info and memoize that info.
+      class SpatialColumnInfo
+        def initialize(adapter, table_name)
+          @adapter = adapter
+          @table_name = table_name
+        end
+
+        def all
+          info = @adapter.query("SELECT f_geometry_column,coord_dimension,srid,type FROM geometry_columns WHERE f_table_name='#{@table_name}'")
+          result = {}
+          info.each do |row|
+            name = row[0]
+            type = row[3]
+            dimension = row[1].to_i
+            has_m = !!(type =~ /m$/i)
```

Comment:
I thought the older `? true : false` is easier to read than this trixy `!!` business.  But if you're only back-porting a change, ignore that I ever said it.